pandas/core/generic.py

```diff
@@ -4455,6 +4455,9 @@ def _align_series(self, other, join='outer', axis=None, level=None,
         raise_on_error : boolean, default True
             Whether to raise on invalid data types (e.g. trying to where on
             strings)
+        apply_other : boolean, default True
+            If False, other will be stored directly rather than applied to
```

Comment:
How about adding internal ``_where`` which doesn't apply callable at all?  ``where`` can use it after applying callable.

I think applying callable to ``other`` is sometimes useful (described in #13299.